src/D2L.Bmx/Aws/AwsCredsCache.cs

```diff
@@ -0,0 +1,83 @@
+using System.Runtime.InteropServices;
+using System.Text.Json;
+namespace D2L.Bmx.Aws;
+
+internal interface IAwsCredentialCache {
+	public AwsCredentials? GetCredentials( string org, string user, AwsRole role, int duration );
+	public void SetCredentials( string org, string user, AwsRole role, AwsCredentials credentials );
+}
+
+internal class AwsCredsCache : IAwsCredentialCache {
+	private static void WriteTextToFile( string path, string content ) {
+		var op = new FileStreamOptions {
+			Mode = FileMode.Create,
+			Access = FileAccess.Write,
+		};
+		if( !RuntimeInformation.IsOSPlatform( OSPlatform.Windows ) ) {
+			op.UnixCreateMode = UnixFileMode.UserRead | UnixFileMode.UserWrite;
+		}
+		using var writer = new StreamWriter( path, op );
+		writer.Write( content );
+	}
+
+	private List<AwsCacheModel> GetAllCache() {
+		if( !File.Exists( BmxPaths.AWS_CREDS_CACHE_FILE_NAME ) ) {
+			return new();
+		}
+		try {
+			string sessionsJson = File.ReadAllText( BmxPaths.AWS_CREDS_CACHE_FILE_NAME );
+			return JsonSerializer.Deserialize( sessionsJson, SourceGenerationContext.Default.ListAwsCacheModel )
+				?? new();
+		} catch( JsonException ) {
+			return new();
+		}
+	}
+
+	void IAwsCredentialCache.SetCredentials( string org, string user, AwsRole role, AwsCredentials credentials ) {
+		List<AwsCacheModel> allEntries = GetAllCache();
+
+		allEntries.Add( new AwsCacheModel(
+			Org: org,
+			User: user,
+			RoleArn: role.RoleArn,
+			Credentials: credentials
+		) );
+
+		var ttlLongerThan = DateTime.UtcNow.AddMinutes( 10 );
+
+		var prunedEntries = allEntries
```

Comment:
Thought about pruning this cache on any BMX use. Its not the greatest look to leave around credentials (granted expired ones) around on someone's home folder.

However without some guards re: "does BMX have write access to write to a .bmx/" etc... it doesn't seem like a good idea to just do that? Especially since unlike the update check process we can't fail this silently.

Reply:
If they're using bmx, they probably also have `~/.aws/credentials` with some expired creds. Sooo... not a big deal I think.